src/github-handler/comment-handler/suggestion-patch-handler/suggestion-hunk-handler.ts

```diff
@@ -0,0 +1,67 @@
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {Hunk, FileHunks, RawChanges, RawContent} from '../../../types';
+import {logger} from '../../../logger';
+import {structuredPatch, Hunk as RawHunk} from 'diff';
+
+/**
+ * Given the old content and new content of a file
+ * compute the diff and extract the necessary hunk data.
+ * The hunk list is a list of disjoint ascending intervals.
+ * @param rawContent
+ * @param fileName
+ * @returns the hunks that are generated in ascending sorted order
+ */
+export function generateHunks(
+  rawContent: RawContent,
+  fileName: string
```

Comment:
Could `RawContent` also track its `fileName`? It simplifies this method conceptually that we're converting from one RawContent change to a list of Hunks